skyportal/tests/api/test_annotations.py

```diff
@@ -0,0 +1,296 @@
+import numpy.testing as npt
+from skyportal.tests import api
+
+
+def test_post_without_origin_fails(annotation_token, public_source, public_group):
+
+    # this should not work, since no "origin" is given
+    status, data = api(
+        'POST',
+        'annotation',
+        data={
+            'obj_id': public_source.id,
+            'data': {'redshift': 0.15},
+            'group_ids': [public_group.id],
+        },
+        token=annotation_token,
+    )
+
+    assert status == 400
+    assert data["message"] == "Missing required field `origin`"
```

Comment:
This will need to be changed when the validation is handled via a Marshmallow schema